src/Libraries/Microsoft.Extensions.AI.Abstractions/Contents/JsonFunctionCallUtilities.cs

```diff
@@ -14,40 +15,68 @@
 using System.Text.Json.Nodes;
 using System.Text.Json.Schema;
 using System.Text.Json.Serialization;
+using System.Text.Json.Serialization.Metadata;
 using System.Text.RegularExpressions;
 using Microsoft.Shared.Diagnostics;
 
-using FunctionParameterKey = (System.Type? Type, string ParameterName, string? Description, bool HasDefaultValue, object? DefaultValue);
+#pragma warning disable S1121 // Assignments should not be made from within sub-expressions
+#pragma warning disable S107 // Methods should not have too many parameters
+
+using FunctionParameterKey = (
+    System.Type? Type,
+    string? ParameterName,
+    string? Description,
+    bool HasDefaultValue,
+    object? DefaultValue,
+    bool IncludeSchemaUri,
+    bool DisallowAdditionalProperties,
+    bool IncludeTypeInEnumSchemas);
 
 namespace Microsoft.Extensions.AI;
 
 /// <summary>Provides a collection of static utility methods for marshalling JSON data in function calling.</summary>
-internal static partial class FunctionCallHelpers
+public static partial class JsonFunctionCallUtilities
 {
+    /// <summary>The uri used when populating the $schema keyword in inferred schemas.</summary>
+#pragma warning disable S1075 // URIs should not be hardcoded
+    private const string SchemaKeywordUri = "https://json-schema.org/draft/2020-12/schema";
+#pragma warning restore S1075 // URIs should not be hardcoded
+
     /// <summary>Soft limit for how many items should be stored in the dictionaries in <see cref="_schemaCaches"/>.</summary>
     private const int CacheSoftLimit = 4096;
 
     /// <summary>Caches of generated schemas for each <see cref="JsonSerializerOptions"/> that's employed.</summary>
     private static readonly ConditionalWeakTable<JsonSerializerOptions, ConcurrentDictionary<FunctionParameterKey, JsonElement>> _schemaCaches = new();
 
     /// <summary>Gets a JSON schema accepting all values.</summary>
-    private static JsonElement TrueJsonSchema { get; } = ParseJsonElement("true"u8);
+    private static readonly JsonElement _trueJsonSchema = ParseJsonElement("true"u8);
 
     /// <summary>Gets a JSON schema only accepting null values.</summary>
-    private static JsonElement NullJsonSchema { get; } = ParseJsonElement("""{"type":"null"}"""u8);
+    private static readonly JsonElement _nullJsonSchema = ParseJsonElement("""{"type":"null"}"""u8);
+
+    /// <summary>
+    /// Removes characters from a .NET member name that shouldn't be used in an AI function name.
+    /// </summary>
+    /// <param name="memberName">The .NET member name that should be sanitized.</param>
+    /// <returns>
+    /// Replaces non-alphanumeric characters in the identifier with the underscore character.
+    /// Primarily intended to remove characters produced by compiler-generated method name mangling.
+    /// </returns>
+    public static string SanitizeMemberName(string memberName) =>
```

Comment:
Nit: now that this is public, we should add a Throw.IfNull(memberName) in there.

Reply:
This is checked by the regex method (added testing for that). I guessing you want this for cleaner stacktraces?

Reply:
> This is checked by the regex method (added testing for that). 

But that exception includes the wrong argument name.

Reply:
Done.